plugins/profiles.koplugin/main.lua

```diff
@@ -57,136 +78,320 @@ function Profiles:getSubMenuItems()
+                end,
+                separator = true,
+            },
+            -- "Edit actions"
+            {
+                text = _("Sort actions"),
+                checked_func = function()
+                    local settings = self.data[k].settings
+                    return settings and settings.actions_order
+                end,
+                callback = function(touchmenu_instance)
+                    self:sortActions(k, touchmenu_instance)
+                end,
+                hold_callback = function(touchmenu_instance)
+                    if self.data[k].settings and self.data[k].settings.actions_order then
+                        self.data[k].settings.actions_order = nil
+                        if #self.data[k].settings == 0 then
+                            self.data[k].settings = nil
+                        end
+                        self.updated = true
+                        touchmenu_instance:updateItems()
                     end
                 end,
+                separator = true,
+            },
+            {
+                text = T(_("Rename: %1"), k),
+                keep_menu_open = true,
+                callback = function(touchmenu_instance)
+                    local function editCallback(new_name)
+                        self.data[new_name] = util.tableDeepCopy(v)
+                        self.data[k] = nil
+                        self.updated = true
+                        self:renameAutostart(k, new_name)
+                        dispatcherRegisterProfile(k, true)
+                        dispatcherRegisterProfile(new_name)
+                        touchmenu_instance.item_table = self:getSubMenuItems()
+                        touchmenu_instance:updateItems()
+                      end
+                    self:editProfileName(editCallback, k)
+                end,
+            },
+            {
+                text = _("Delete"),
+                keep_menu_open = true,
+                separator = true,
+                callback = function(touchmenu_instance)
+                    UIManager:show(ConfirmBox:new{
+                        text = _("Do you want to delete this profile?"),
+                        ok_text = _("Delete"),
+                        ok_callback = function()
+                            self.data[k] = nil
+                            self.updated = true
+                            self:renameAutostart(k)
+                            dispatcherRegisterProfile(k, true)
+                            touchmenu_instance.item_table = self:getSubMenuItems()
+                            touchmenu_instance:updateItems()
+                        end,
+                    })
+                end,
             },
         }
-        Dispatcher:addSubMenu(self, sub_items, self.data, k)
+        local edit_actions_sub_items = {}
+        Dispatcher:addSubMenu(self, edit_actions_sub_items, self.data, k)
+        table.insert(sub_items, 5, { 
```

Comment:
(I'm not having a great look at the whole context, but any reason to table.insert it here at the specific 5th position, vs. doing it inline like all other menu items ?)